packages/composites/token-allocation/src/adapter.ts

```diff
@@ -159,6 +168,19 @@ export const execute = async (input: AdapterRequest, config: Config): Promise<Ad
   }
 }
 
+const validateEndpointSupportedBySource = (jobRunID: string, source: Source, method: string) => {
+  source = source.toUpperCase()
+
+  // Todo:  In the future we should aim to export the supported endpoints from EAs
+  if (method === 'marketcap' && (source === NCFX.NAME || source === Finage.NAME)) {
```

Comment:
Hmm would this be better as a negative list? Wondering about the possibility of extending `token-allocation` to other adapters in the future.
```
const adaptersSupportingMarketcap = [ ... ]
...
if (method === 'marketcap' && !adaptersSupportingMarketcap.includes(source)) {
```